### PR TITLE
Show quickfix after compile & Introduce manual code cleanup action

### PR DESCRIPTION
**Pull request by asmodeus812**
Make compile and build more robust, make error reporting more verbose, show quickfix list with the error entries. Add a newly introduced jdtls clean up action that applies manual cleanup.

### CODE REVIEW

File: src/standardLanguageClient.ts [hunk 1]
```diff
@@ -471,28 +474,29 @@ export class StandardLanguageClient {
...
-          const humanVisibleDelay = elapsed < 1000 ? 1000 : 0
-          return new Promise((resolve, reject) => {
+            const elapsed = new Date().getTime() - start
+            const humanVisibleDelay = elapsed < 1000 ? 1000 : 0
+
+            if (res == CompileWorkspaceStatus.withError) {
+              showCompileBuildDiagnostics()
+              window.showWarningMessage("Compilation finished with errors")
+            }
             setTimeout(() => { // set a timeout so user would still see the message when build time is short
```
Comment (asmodeus812) [inline src/standardLanguageClient.ts:498]: There was a bug with this reject, the progress for compile was stuck on Compiling... passed result to resolve

**Comment by asmodeus812** [comment]
Resolves https://github.com/neoclide/coc-java/issues/283

**Comment by asmodeus812** [comment]
@chemzqm could you take a look, thanks

**Comment by sainnhe** [comment]
So sad that this PR gets closed. Is there any chance to reopen this PR and let it merged?

**Comment by asmodeus812** [comment]
@sainnhe my guy, if you take a look at my fork, it is already ahead by two commits - https://github.com/asmodeus812/coc-java. Those include the changes to the action amongst other things, my extension is published under `coc-java-dev` in npm registry, if you wish you can use that.

**Comment by sainnhe** [comment]
Great work, thanks! I'm going to look at it.

**Pull request closed without merging**